controllers/nodehealthcheck_controller_test.go

```diff
@@ -315,6 +315,34 @@ var _ = Describe("Node Health Check CR", func() {
 			})
 		})
 
+		Context("control plane nodes", func() {
+			When("two control plane nodes are unhealthy, just one should be remediated", func() {
+				BeforeEach(func() {
+					objects = newNodes(2, 1, true)
+					objects = append(objects, newNodes(1, 5, false)...)
+					underTest = newNodeHealthCheck()
+					remediationTemplate := newRemediationTemplate()
+					objects = append(objects, underTest, remediationTemplate)
+				})
+
+				It("creates a one remediation CR for control plane node", func() {
+					Expect(reconcileError).NotTo(HaveOccurred())
+
+					cr := newRemediationCR("")
+					crList := &unstructured.UnstructuredList{Object: cr.Object}
+					Expect(reconciler.Client.List(context.Background(), crList)).To(Succeed())
+
+					Expect(len(crList.Items)).To(BeNumerically("==", 2), "expected 2 remediations, one for control plane, one for worker")
+					Expect(crList.Items).To(ContainElements(
+						// the unhealthy worker
+						HaveField("Object", HaveKeyWithValue("metadata", HaveKeyWithValue("name", "unhealthy-node-1"))),
+						// one of the unhealthy control plane nodes
+						HaveField("Object", HaveKeyWithValue("metadata", HaveKeyWithValue("name", ContainSubstring("unhealthy-control-plane-node")))),
```

Comment:
In line 603 we add an integer suffix to the node name, thus shouldn't the node name be `unhealthy-control-plane-node-1`?
In line 308 we do have the number suffix ( "unhealthy-node-**1**")

Reply:
we have 2 unhealthy control plane nodes, and don't know which one is remediated, "-1" or "-2"

Reply:
Do you mean there could be a race condition between them so possibly `unhealthy-control-plane-node-2` will be remediated first?

Reply:
I wouldn't call it race condition, because AFAIK that's about unwanted / bad behaviour. Here it's just undefined and unimportant which one gets remediated first. The important thing is that it's one only.